ocs_ci/ocs/longevity_stage2.py

```diff
@@ -0,0 +1,563 @@
+    for index, start_time in enumerate(time_measures):
+        logger.info(
+            f"{msg_prefix} pod number {index + 1} start time: {start_time} seconds"
+        )
+        if start_time > 30:
+            raise ex.PerformanceException(
+                f"{msg_prefix} Pod number {index} start time is {start_time},"
+                f"which is greater than 30 seconds"
+            )
+    if time_measures:
+        average = statistics.mean(time_measures)
+        logger.info(
+            f"{msg_prefix} The average time for the sampled {len(time_measures)} pods is {average} seconds."
+        )
+
+
+def measure_pod_creation_time(namespace, num_of_pods):
+    """
+    Measures and Logs the POD Creation Time of all the PODs.
+
+    Args:
+        namespace (str) : Namespace in which the PODs are created.
+        num_of_pods (int) : Number of PODs created.
+
+    Raises:
+        PerformanceException : Raises an exception if POD creation time is greater than the accepted time.
+
+    Logs:
+        POD Creation Time of all the PODs.
+
+    """
+    logs = performance_lib.run_oc_command(
+        "get events --sort-by='{.lastTimestamp}'",
+        namespace,
+    )
+
+    scheduled_time = None
+    pod_no = num_of_pods
+    accepted_creation_time = 12
+
+    for line in logs:
+        logger.info(line)
+        if "Scheduled" in line:
+            scheduled_time = int(line.split()[0][:-1])
+        elif "Created" in line:
+            created_time = int(line.split()[0][:-1])
+            creation_time = scheduled_time - created_time
+            logger.info(f"POD number {pod_no} was created in {creation_time} seconds.")
+            if creation_time > accepted_creation_time:
+                raise ex.PerformanceException(
+                    f"POD creation time is {creation_time} and is greater than "
+                    f"{accepted_creation_time} seconds."
+                )
+            pod_no -= 1
+
+
+def measure_pvc_creation_time(interface, pvc_size, pvc_objs, start_time):
+    """
+    Measures and Logs PVC Creation Time of all PVCs.
+
+    Args:
+        interface (str) : an interface (RBD or CephFS) to run on.
+        pvc_size (int) : Size of each PVC in GB.
+        pvc_objs (list) : List of PVC objects for which we have to measure the time.
+        start_time (str) : Formatted time from which and on to search the relevant logs.
+
+    Raises:
+        PerformanceException : Raises an exception if PVC creation time is greater than the accepted time.
+
+    Logs:
+        PVC Creation Time of all the PVCs.
+
+    """
+    msg_prefix = f"Interface: {interface}, PVC size: {pvc_size}."
+    accepted_creation_time = 1
+    pvc_no = 0
+    for pvc_obj in pvc_objs:
+        creation_time = performance_lib.measure_pvc_creation_time(
+            interface, pvc_obj.name, start_time
+        )
+
+        logger.info(
+            f"{msg_prefix} PVC number {pvc_no + 1} of interface {interface} was created in {creation_time} seconds."
```

Comment:
Instead of the pvc_number , you can directly log the pvc name using the pvc_obj. that way it will be more helpful

Reply:
Ack